src/Tests.Microsoft.Security.Utilities.Core/DetectionTests.cs

```diff
@@ -208,13 +208,13 @@ public void Detection_EqualComparisonUpdateWhenPropertiesChange()
             previousDetection = new Detection(currentDetection);
             Assert.AreEqual(currentDetection, previousDetection);
 
-            currentDetection.Start = Math.Max(1, (int)DateTime.UtcNow.Ticks % 99);
+            currentDetection.Start++;
```

Comment:
>Start [](http://example.com/codeflow?start=29&length=5)

There have been several poorly thought out attempts to make this test not occasionally fail due to randomly matching data fixed earlier in execution. This is the latest poorly thought out attempt.

I think it's right this time? Basically we constrain all data for these properties in a range from 0-98 and for all values it is both safe to increment them by one and by doing so we should invalidate equality and pass the test.

Reply:
I wanted to understand what this test is doing to review this. It points to a larger design issue with mutable properties of a reference type factoring in GetHashCode()/Equals(). This is a footgun that can cause objects to be lost in dictionaries.

See https://learn.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-9.0#notes-to-inheritors

Why is Detection mutable?

Reply:
#169